ironic/data_source_ironic_introspection.go

```diff
@@ -79,11 +80,26 @@ func dataSourceIronicIntrospectionRead(d *schema.ResourceData, meta interface{})
 		return fmt.Errorf("could not get introspection status: %s", err.Error())
 	}
 
-	d.Set("finished", status.Finished)
-	d.Set("finished_at", status.FinishedAt)
-	d.Set("started_at", status.StartedAt)
-	d.Set("error", status.Error)
-	d.Set("state", status.State)
+	err = d.Set("finished", status.Finished)
+	if err != nil {
+		return err
+	}
+	err = d.Set("finished_at", status.FinishedAt.Format("2006-01-02T15:04:05"))
```

Comment:
this never worked as there was a type error (setting time.Time to a string field)